master/internal/root.go

```diff
@@ -25,17 +16,21 @@ func root(ctx *actor.Context) error {
 	case actor.ChildFailed:
 		switch msg.Child.Address() {
 		case sproto.K8sRMAddr, sproto.AgentRMAddr:
-			os.Exit(exit(ctx))
+			ctx.Log().WithField("crash", ctx.Message()).Errorf(clusterCrashMessage)
+			return ctx.Self().AwaitTermination()
```

Comment:
if we await our own termination, we will never exit.
```suggestion
			return errors.New(clusterCrashMessage)
```
better yet, just change `clusterCrashMessage` to an `error`